pkg/input/cloudapi/client.go

```diff
@@ -0,0 +1,74 @@
+package cloudapi
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"regexp"
+)
+
+type ClientConfig struct {
+	URL     string
+	Token   string
+	Version string
+}
+
+type Client struct {
+	httpClient    *http.Client
+	url           string
+	authorization string
+	version       string
+}
+
+func NewClient(config ClientConfig) (*Client, error) {
+	if config.Token == "" {
+		return nil, fmt.Errorf("no token provided")
+	}
+
+	u := config.URL
+	if u == "" {
+		u = "https://api.snyk.io"
```

Comment:
Nitpick: move this and the version to the top of this file as a const or var

Reply:
Thanks! Fixed in 981fed0.